x-pack/legacy/plugins/siem/public/components/matrix_histogram/index.tsx

```diff
@@ -73,32 +80,54 @@ const getBarchartConfigs = (from: number, to: number, onBrushEnd: UpdateDateRang
   customHeight: 324,
 });
 
-export const MatrixOverTimeHistogram = ({
-  customChartData,
-  id,
-  loading,
+const formatToChartDataItem = ([key, value]: [
+  string,
+  MatrixHistogramDataTypes[]
+]): ChartSeriesData => ({
+  key,
+  value,
+});
+
+const getCustomChartData = (
```

Comment:
maybe we could extract logic to `./utils.ts`?

Reply:
Good call, isolated utils functions and types to separate files.